pkg/cache/application.go

```diff
@@ -396,7 +438,7 @@ func (app *Application) Schedule() bool {
 }
 
 func (app *Application) scheduleTasks(taskScheduleCondition func(t *Task) bool) {
-	for _, task := range app.GetNewTasks() {
+	for _, task := range app.newTasks.getTasks() {
```

Comment:
There is one thing that the patch doesn't take into account: the order of pods based on submission time. I'll update it on Monday.

Reply:
Ok, I couldn't wait so I updated it already.